src/lib/proposer/proposer.ml

```diff
@@ -350,96 +356,107 @@ let run ~logger ~prover ~verifier ~trust_system ~get_completed_work
                              `This_transition_was_generated_internally
                         |> Transition_frontier_validation
                            .validate_frontier_dependencies ~logger ~frontier
-                        |> Result.map_error ~f:(fun err ->
-                               let exn name =
-                                 Error.to_exn
-                                   (Error.of_string
+                      with
+                      | Error `Already_in_frontier ->
+                          Logger.error logger ~module_:__MODULE__
```

Comment:
Some extra metadata for these logs might be nice? Perhaps The generated `protocol_state`. If we see these logs happen during the testnet, I'm not sure we'd have enough information to understand and fix the bug.